Kafka bug in release 1.2.0 and master

Hello,

I have made an error when I have merged my Kafka TLS feature with your change on plugin registration.

Use case : Run with : ` --input-raw :8000 --output-kafka-host mykafka:36667 --output-kafka-topic mytopic`
```
panic: reflect: Call with too many input arguments

goroutine 1 [running]:
reflect.Value.call(0xc8fbc0, 0xe16b20, 0x13, 0xdc2813, 0x4, 0xc000093560, 0x3, 0x4, 0x1, 0x0, ...)
        /usr/lib/golang/src/reflect/value.go:380 +0x134c
reflect.Value.Call(0xc8fbc0, 0xe16b20, 0x13, 0xc000093560, 0x3, 0x4, 0x2, 0x4, 0x1)
        /usr/lib/golang/src/reflect/value.go:321 +0xb4
main.(*InOutPlugins).registerPlugin(0xc000094910, 0xc8fbc0, 0xe16b20, 0xc00029fda8, 0x3, 0x3)
        /media/data/POC/goreplay/plugins.go:50 +0x1f4
main.NewPlugins(0xc0000922a0)
        /media/data/POC/goreplay/plugins.go:140 +0x52f
main.main()
        /media/data/POC/goreplay/gor.go:53 +0x8f2

Process finished with exit code 2
```

Error is here : I keep registerPlugin in replacement of plugins.registerPlugin but I miss the change of NewKafkaOutputWithTLS (function with 3 args instead of 2).
It is not visible in Tests because I keep the old function named NewKafkaOutput and redirect on mine.

So the error in 1.2.0 and master :
```
	if Settings.OutputKafkaConfig.Host != "" && Settings.OutputKafkaConfig.Topic != "" {
		registerPlugin(NewKafkaOutput, "", &Settings.OutputKafkaConfig, &Settings.KafkaTLSConfig)
	}

	if Settings.InputKafkaConfig.Host != "" && Settings.InputKafkaConfig.Topic != "" {
		registerPlugin(NewKafkaInput, "", &Settings.InputKafkaConfig, &Settings.KafkaTLSConfig)
	}
```

Has to be corrected in : 
```
	if Settings.OutputKafkaConfig.Host != "" && Settings.OutputKafkaConfig.Topic != "" {
		registerPlugin(NewKafkaOutputWithTLS, "", &Settings.OutputKafkaConfig, &Settings.KafkaTLSConfig)
	}

	if Settings.InputKafkaConfig.Host != "" && Settings.InputKafkaConfig.Topic != "" {
		registerPlugin(NewKafkaInputWithTLS, "", &Settings.InputKafkaConfig, &Settings.KafkaTLSConfig)
	}
```

--- a/input_kafka.go
+++ b/input_kafka.go
@@ -17,13 +17,8 @@ type KafkaInput struct {
 	messages  chan *sarama.ConsumerMessage
 }
 
-// NewKafkaInput creates instance of kafka consumer client.
-func NewKafkaInput(address string, config *InputKafkaConfig) *KafkaInput {
-	return NewKafkaInputWithTLS(address, config, nil)
-}
-
-// NewKafkaInputWithTLS creates instance of kafka consumer client with TLS
-func NewKafkaInputWithTLS(address string, config *InputKafkaConfig, tlsConfig *KafkaTLSConfig) *KafkaInput {
+// NewKafkaInput creates instance of kafka consumer client with TLS config
+func NewKafkaInput(address string, config *InputKafkaConfig, tlsConfig *KafkaTLSConfig) *KafkaInput {
 	c := NewKafkaConfig(tlsConfig)
 
 	var con sarama.Consumer

--- a/input_kafka_test.go
+++ b/input_kafka_test.go
@@ -20,7 +20,7 @@ func TestInputKafkaRAW(t *testing.T) {
 		consumer: consumer,
 		Topic:    "test",
 		UseJSON:  false,
-	})
+	},nil)
 
 	buf := make([]byte, 1024)
 	n, err := input.Read(buf)
@@ -47,7 +47,7 @@ func TestInputKafkaJSON(t *testing.T) {
 		consumer: consumer,
 		Topic:    "test",
 		UseJSON:  true,
-	})
+	},nil)
 
 	buf := make([]byte, 1024)
 	n, err := input.Read(buf)

--- a/output_kafka.go
+++ b/output_kafka.go
@@ -23,13 +23,8 @@ type KafkaOutput struct {
 // KafkaOutputFrequency in milliseconds
 const KafkaOutputFrequency = 500
 
-// NewKafkaOutput creates instance of kafka producer client.
-func NewKafkaOutput(address string, config *OutputKafkaConfig) io.Writer {
-	return NewKafkaOutputWithTLS(address, config, nil)
-}
-
-// NewKafkaOutputWithTLS creates instance of kafka producer client.
-func NewKafkaOutputWithTLS(address string, config *OutputKafkaConfig, tlsConfig *KafkaTLSConfig) io.Writer {
+// NewKafkaOutput creates instance of kafka producer client  with TLS config
+func NewKafkaOutput (address string, config *OutputKafkaConfig, tlsConfig *KafkaTLSConfig) io.Writer {
 	c := NewKafkaConfig(tlsConfig)
 
 	var producer sarama.AsyncProducer

--- a/output_kafka_test.go
+++ b/output_kafka_test.go
@@ -17,7 +17,7 @@ func TestOutputKafkaRAW(t *testing.T) {
 		producer: producer,
 		Topic:    "test",
 		UseJSON:  false,
-	})
+	},nil)
 
 	output.Write([]byte("1 2 3\nGET / HTTP1.1\r\nHeader: 1\r\n\r\n"))
 
@@ -40,7 +40,7 @@ func TestOutputKafkaJSON(t *testing.T) {
 		producer: producer,
 		Topic:    "test",
 		UseJSON:  true,
-	})
+	}, nil)
 
 	output.Write([]byte("1 2 3\nGET / HTTP1.1\r\nHeader: 1\r\n\r\n"))
 

